wire-schema/src/test/java/com/squareup/wire/schema/SchemaTypeAdapterTest.java

```diff
@@ -77,8 +80,78 @@
 
   @Test public void typeAdapterDecode() throws Exception {
     MessageType cafeDrink = (MessageType) coffeeSchema.getType("CafeDrink");
-    MessageAdapter<Map<String, Object>> adapter = typeAdapterFactory.get(cafeDrink);
+    MessageAdapter<Map<String, Object>> adapter = typeAdapterFactory.get(cafeDrink.name());
     assertThat(adapter.read(new Buffer().write(dansCoffeeEncoded))).isEqualTo(dansCoffee);
     assertThat(adapter.read(new Buffer().write(jessesCoffeeEncoded))).isEqualTo(jessesCoffee);
   }
+
+  @Test public void groupsIgnored() throws IOException {
```

Comment:
I like that this structure allows the tests to be self-contained. But it's also a bit weird to have general parser tests here.